### PR TITLE
Add sensible extension

### PR DESCRIPTION
**Pull request by michaelwschultz**
## Description

Adds ability to people to extract data from their documents using [Sensible.so](https://sensible.so) right from Raycast.
Requires a free Sensible account + API key to get started. (Instructions in the readme)

## Screencast

[Demo on Youtube](https://youtu.be/MDDjYnzpjOE)

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by michaelwschultz** [comment]
Looks like our new Sensible SDK isn't exporting all the types correctly. I'll fix that on our side and update the SDK dependency to fix the issue tomorrow.

**Comment by michaelwschultz** [comment]
Alright, this should be ready for review now.

File: extensions/sensible/package.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      "title": "Run Extraction",
+      "subtitle": "Sensible",
+      "description": "Extract structured data from documents using Sensible.so",
+      "mode": "view"
+    }
+  ],
+  "preferences": [
+    {
+      "name": "sensible_api_key",
+      "title": "Sensible API key",
```
Comment (pernielsentikaer) [inline extensions/sensible/package.json:24]:
```suggestion
      "title": "Sensible API Key",
```

File: extensions/sensible/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  return (
+    <Form
+      isLoading={isLoadingDocumentTypes}
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm title="Submit" onSubmit={handleSubmit} />
+        </ActionPanel>
+      }
+    >
+      <Form.Dropdown title="Document type" {...itemProps.documentType}>
```
Comment (pernielsentikaer) [inline extensions/sensible/src/index.tsx:40]:
```suggestion
      <Form.Dropdown title="Document Type" {...itemProps.documentType}>
```

File: extensions/sensible/src/types.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export type DocumentType = {
+  id: string;
+  name: string;
+  created: string;
+};
+
+export interface Preferences {
+  sensible_api_key: string;
+}
```
Comment (pernielsentikaer) [inline extensions/sensible/src/types.ts:9]:
```suggestion
```
This is already build into Raycast with `ExtensionPreferences`

File: extensions/sensible/src/hooks/useDocumentTypes.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { getPreferenceValues } from "@raycast/api";
+import { useFetch } from "@raycast/utils";
+
+import type { DocumentType, Preferences } from "../types";
```
Comment (pernielsentikaer) [inline extensions/sensible/src/hooks/useDocumentTypes.ts:4]:
```suggestion
import type { DocumentType } from "../types";
```

File: extensions/sensible/src/hooks/useDocumentTypes.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+const ENV = "v0";
+
+export default function useDocumentTypes() {
+  const { sensible_api_key } = getPreferenceValues<Preferences>();
```
Comment (pernielsentikaer) [inline extensions/sensible/src/hooks/useDocumentTypes.ts:9]:
```suggestion
  const { sensible_api_key } = getPreferenceValues<ExtensionPreferences>();
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/sensible/src/types.ts (hunk 3 above)
Comment (michaelwschultz) [reply extensions/sensible/src/types.ts:9]: Oops! Forgot about that. Thanks!

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**